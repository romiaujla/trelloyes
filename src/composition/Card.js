import React from 'react';

class Card extends React.Component {

    static defaultProps = {
        id: ``,
        listId: ``,
        title: ``,
        handleDeleteButtonClick: () => {}
    }
    
    render(){
        const props = this.props;
        return (
            <li className="card">
                <div className="flex-div">
                    <h3>{props.title}</h3>
                    <button type="button" className="card-delete-button" onClick={() => this.props.handleDeleteButtonClick(this.props.id, this.props.listId)}>delete</button>
                </div>
                <p>{props.content}</p>
            </li>
        );
    }
}

export default Card;