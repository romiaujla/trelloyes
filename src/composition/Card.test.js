import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

describe('Card Component', ()=>{
    
    it('renders without crashing', ()=>{
        const ul = document.createElement('ul');
        ReactDOM.render(<Card />, ul);
        ReactDOM.unmountComponentAtNode(ul);
    });

    it('renders with title and content', ()=>{
        const tree = renderer
            .create(<Card title="Second card" content="lorem ipsum" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});