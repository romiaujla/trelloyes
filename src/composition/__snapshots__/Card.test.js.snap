// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders with title and content 1`] = `
<li
  className="card"
>
  <div
    className="flex-div"
  >
    <h3>
      Second card
    </h3>
    <button
      className="card-delete-button"
      onClick={[Function]}
      type="button"
    >
      delete
    </button>
  </div>
  <p>
    lorem ipsum
  </p>
</li>
`;
